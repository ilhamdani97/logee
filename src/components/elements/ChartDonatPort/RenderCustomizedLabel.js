import React from 'react';
import styles from './styles.scoped.css';

const RenderCustomizedLabel = ({
  cx,
  cy,
  midAngle,
  innerRadius,
  outerRadius,
  percent,
}) => {
  const RADIAN = Math.PI / 180;
  const radius = innerRadius + (outerRadius - innerRadius) * 0.5;
  const x = cx + radius * Math.cos(-midAngle * RADIAN);
  const y = cy + radius * Math.sin(-midAngle * RADIAN);

  const renderCircle = (<g className={styles.circleFill}>
    <circle
      cx={x > cx ? x+10 : x-10}
      cy={y > cy ? y+10 : y-10}
      r="15" />
    <text
      className={styles.textCircle}
      dominantBaseline="central"
      fill="black"
      textAnchor="middle"
      x={x > cx ? x+10 : x-10}
      y={y > cy ? y+10 : y-10}
    >
      {`${(percent * 100).toFixed(0)}%`}
    </text>
  </g>);

  return percent !== 0 ? renderCircle :null;
};

export default RenderCustomizedLabel;
