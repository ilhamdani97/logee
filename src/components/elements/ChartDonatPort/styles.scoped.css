.circleFill{
  fill: #f5f5f5;
}

.responsiveChart{
  width: 100%;
  height: 300px;
}
.textCircle{
  font-size: 10px;
}