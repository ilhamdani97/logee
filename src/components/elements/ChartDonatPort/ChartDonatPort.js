import React from 'react';
import { PieChart, Pie, Cell, ResponsiveContainer } from 'recharts';
import styles from './styles.scoped.css';
import PropTypes from 'prop-types';
import RenderCustomizedLabel from './RenderCustomizedLabel';


const ChartDonatPort = ({ data, colors }) => {

  return (
    <div className={styles.responsiveChart}>
      <ResponsiveContainer>
        <PieChart>
          <Pie
            cornerRadius={100}
            data={data}
            dataKey="value"
            endAngle={-180}
            innerRadius={85}
            label={<RenderCustomizedLabel />}
            labelLine={false}
            outerRadius={110}
            paddingAngle={-10}
            startAngle={180}
            stroke="none"
          >
            {data.map((_, index) => (
              <Cell fill={colors[index % colors.length]} key={`cell-${index}`} />
            ))}
          </Pie>
        </PieChart>
      </ResponsiveContainer>
    </div>
  );
};



ChartDonatPort.defaultProps = {
  colors:[],
  data: [],
};

ChartDonatPort.propTypes = {
  colors:PropTypes.array,
  data: PropTypes.array,
};


export default ChartDonatPort;
