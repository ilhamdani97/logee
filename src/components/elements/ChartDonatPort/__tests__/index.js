import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import ChartDonatPort from '../ChartDonatPort';

const data =  [
  {
    'name': 'totalOnProcess',
    'value': 0
  },
  {
    'name': 'totalOnConfirm',
    'value': 2
  },
  {
    'name': 'totalOnDelivery',
    'value': 1
  },
  {
    'name': 'totalDone',
    'value': 11
  },
  {
    'name': 'totalExpired',
    'value': 0
  }
];

const colors =  [
  '#FAB005',
  '#56C281',
  '#064C6F',
  '#8DD4F7',
  '#DE1B1B'
];

describe('src/components/elements/ChartDonatPort', () => {
  test('render', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<ChartDonatPort colors={colors} data={data}/>);
    expect(tree).toMatchSnapshot();
  });
});
