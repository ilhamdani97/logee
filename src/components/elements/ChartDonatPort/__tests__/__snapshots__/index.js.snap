// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/elements/ChartDonatPort render 1`] = `
<div>
  <ForwardRef>
    <PieChart
      barCategoryGap="10%"
      barGap={4}
      cx="50%"
      cy="50%"
      endAngle={360}
      innerRadius={0}
      layout="centric"
      margin={
        Object {
          "bottom": 5,
          "left": 5,
          "right": 5,
          "top": 5,
        }
      }
      outerRadius="80%"
      reverseStackOrder={false}
      stackOffset="none"
      startAngle={0}
      syncMethod="index"
    >
      <Pie
        animationBegin={400}
        animationDuration={1500}
        animationEasing="ease"
        blendStroke={false}
        cornerRadius={100}
        cx="50%"
        cy="50%"
        data={
          Array [
            Object {
              "name": "totalOnProcess",
              "value": 0,
            },
            Object {
              "name": "totalOnConfirm",
              "value": 2,
            },
            Object {
              "name": "totalOnDelivery",
              "value": 1,
            },
            Object {
              "name": "totalDone",
              "value": 11,
            },
            Object {
              "name": "totalExpired",
              "value": 0,
            },
          ]
        }
        dataKey="value"
        endAngle={-180}
        fill="#808080"
        hide={false}
        innerRadius={85}
        isAnimationActive={true}
        label={<RenderCustomizedLabel />}
        labelLine={false}
        legendType="rect"
        minAngle={0}
        nameKey="name"
        outerRadius={110}
        paddingAngle={-10}
        startAngle={180}
        stroke="none"
      >
        <Cell
          fill="#FAB005"
        />
        <Cell
          fill="#56C281"
        />
        <Cell
          fill="#064C6F"
        />
        <Cell
          fill="#8DD4F7"
        />
        <Cell
          fill="#DE1B1B"
        />
      </Pie>
    </PieChart>
  </ForwardRef>
</div>
`;
